gk/main.c

```diff
@@ -1037,9 +1037,9 @@ print_flow_state(struct flow_entry *fe)
 		RTE_BUILD_BUG_ON(RTE_DIM(fe->u.bpf.cookie.mem) != 8);
 
 		ret = snprintf(state_msg, sizeof(state_msg),
-			"gk: log the flow state [state: GK_BPF (%hhu), expire_at: %"PRIx64", program_index=%u, cookie="
-			"%16" PRIx64 "%16" PRIx64 "%16" PRIx64 "%16" PRIx64
-			"%16" PRIx64 "%16" PRIx64 "%16" PRIx64 "%16" PRIx64 ", grantor_ip: %s] in the flow table at %s with lcore %u",
+			"gk: log the flow state [state: GK_BPF (%hhu), expire_at: 0x%"PRIx64", program_index=%u, cookie="
+			"0x%16" PRIx64 "0x%16" PRIx64 "0x%16" PRIx64 "0x%16" PRIx64
+			"0x%16" PRIx64 "0x%16" PRIx64 "0x%16" PRIx64 "0x%16" PRIx64 ", grantor_ip: %s] in the flow table at %s with lcore %u",
```

Comment:
1. Wouldn't it be better to turn `%16` into `%016` instead of adding the prefix `0x` to each 64 bits?

2. Whatever choice we make above, update `print_unsent_policy()` accordingly.

Reply:
If needed, update the title of the patch and of the pull request.

Reply:
Another thing, use `rte_cpu_to_be_64()` on the values passed to `snprintf()` so the bytes are printed in the order they are stored in the memory.

Reply:
I know that we leave it up to the BPF program to decide what to do with the cookie's byte ordering, but I don't see why we should do a conversion here. Converting from CPU --> networking ordering seems to go against the goal of printing the bytes in the order they are stored in memory.

Reply:
The assumption you're making, namely, that we are already printing the bytes in the order they are stored in memory is **not** true. To see it for yourself, fill up a cookie with bytes from 1 to 64 and print them out. The conversion I'm proposing will put them in the proper order.

Reply:
> 2. Whatever choice we make above, update `print_unsent_policy()` accordingly.

You've forgotten to update `print_unsent_policy()`.

Reply:
Oops -- should be all set now.